linode/terraform-provider-linode
# Avoid using resource attributes in data source schemas for VPC subnet

## 📝 Description

A data sources schema shouldn't use attributes from the resource schema package (`github.com/hashicorp/terraform-plugin-framework/resource/schema`) and should use attributes from data source schema package (`github.com/hashicorp/terraform-plugin-framework/datasource/schema`) instead.

This PR is for correcting them.

There is a [feature request](https://github.com/hashicorp/terraform-plugin-framework/issues/568) and a [community effort](https://github.com/hashicorp/terraform-plugin-framework/issues/568#issuecomment-1491896510) of implementing a plugin to support shared schema between data sources and resources, but there is not a plan to implement it in the official Terraform Framework package.

## ✔️ How to Test

### Automated testing
`make testacc PKG_NAME="linode/vpcsubnets"`
`make testacc PKG_NAME="linode/vpcsubnet"`

GHA Runs:
https://github.com/linode/terraform-provider-linode/actions/runs/7245168792
https://github.com/linode/terraform-provider-linode/actions/runs/7245166453

### Manual testing

```terraform
resource "linode_vpc" "test" {
    label = "test-vpc"
    region = "us-mia"
    description = "My first VPC."
}

resource "linode_vpc_subnet" "test" {
    vpc_id = linode_vpc.test.id
    label = "test-subnet"
    ipv4 = "10.0.0.0/24"
}

data "linode_vpc_subnets" "filtered-subnets" {
    vpc_id = linode_vpc.test.id
}

data "linode_vpc_subnet" "foo" {
    vpc_id = linode_vpc.test.id
    id = linode_vpc_subnet.test.id
}

output "vpc_subnet" {
  value = data.linode_vpc_subnet.foo
}

output "vpc_subnets" {
  value = data.linode_vpc_subnets.filtered-subnets.vpc_subnets
}
```

Run `terraform apply --auto-approve`, and you can see the output from `linode_vpc_subnet`.
Then run `terraform refresh`, you should be able to see the output from `linode_vpc_subnets`
